src/Decompiler/Analysis/DataFlowAnalysis.cs

```diff
@@ -276,6 +276,14 @@ public SsaTransform ConvertToSsa(Procedure proc)
             sst.RenameFrameAccesses = true;
             sst.Transform();
 
+            var icrw = new IndirectCallRewriter(program, ssa, eventListener);
+            while (icrw.Rewrite())
+            {
```

Comment:
Please add a check that `eventListener.IsCanceled()` is false in the while-loop.